tests/test_styler.py

```diff
@@ -63,6 +63,18 @@ def f1(s: Series) -> Series[str]:
     check(assert_type(DF.style.apply_index(f1), Styler), Styler)
 
 
+def test_map_index() -> None:
+    def f(s: Series) -> npt.NDArray[np.str_]:
+        return np.asarray(s, dtype=np.str_)
+
+    check(assert_type(DF.style.map_index(f), Styler), Styler)
+
+    def f1(s: Series) -> Series[str]:
+        return Series(s, dtype=str)
+
+    check(assert_type(DF.style.map_index(f1), Styler), Styler)
```

Comment:
The tests should be modified to something similar to what you see in the docs for `map_index()`

Reply:
Updated to show the first example from the docs. The second one showed the following error

```
    def f1(s: Scalar):
        return "background-color: yellow;" if "x" in s else None
                                              ~~~~~~~~
Unsupported right operand type for in ("str | bytes | date | datetime | timedelta | datetime64 | timedelta64 | bool | int | float | Timestamp | Timedelta | complex")
```